src/vs/editor/common/config/commonEditorConfig.ts

```diff
@@ -398,7 +383,7 @@ export abstract class CommonEditorConfiguration extends Disposable implements IC
 					continue;
 				}
 				if (baseValue && typeof baseValue === 'object' && subsetValue && typeof subsetValue === 'object') {
-					if (!this._subsetEquals(baseValue, subsetValue)) {
+					if (!objects.equals(baseValue, subsetValue)) {
```

Comment:
I don't think this change preserves existing semantics. Here's an example on how `_subsetEquals` was working before:

```
editor.updateOptions({ minimap: { enabled: true, showSlider: 'always' } });
editor.updateOptions({ minimap: { enabled: false } }); // effective value: { enabled: false, showSlider: 'always' }
editor.updateOptions({ minimap: { enabled: false } }); // effective value: { enabled: false, showSlider: 'always' }
```

`_subsetEquals` was supposed to make the 3rd call be a no-op. But if I understand correctly the proposed changes, the third call will now go through because `object.equals({ enabled: false, showSlider: 'always' }, { enabled: false })` returns `false`.